src/metrics/representation.py

```diff
@@ -195,18 +195,22 @@ def prepare_representation_data(self, vectors: np.ndarray, queries_idxs: np.ndar
         
         relevant = []
         empty_relevant_idxs = []
+        # print(f'scores = {scores}')
         for idx in range(len(q_vecs)):
             relevant_idxs = np.where(scores[:, queries_idxs[idx]] > 0.)[0]
             if len(relevant_idxs) == 0:
                 empty_relevant_idxs.append(idx)
             else:
+                # Need sort relevant indexes by its scores for NDCG metric
+                current_scores = scores[relevant_idxs, queries_idxs[idx]]
+                sort_indexes = np.argsort(current_scores)
+                relevant_idxs = relevant_idxs[sort_indexes[::-1]]
                 relevant.append(relevant_idxs)
-        relevant = np.array(relevant)
-        
+        relevant = np.array(relevant, dtype=np.object)
```

Comment:
Not a good type. Why is it like that?

Reply:
It is some warning numpy just try it fix, but unsuccessfully, just forgot delete.